File: 400/458 - Poor Pigs.java
public class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int len = minutesToTest / minutesToDie + 1, pigs = 0, prod = 1;
        
        while (prod < buckets) {
            prod *= len;
            ++pigs;
        }
        
        return pigs;
    }
}

File: csharp/458.poor-pigs.cs
/*
 * @lc app=leetcode id=458 lang=csharp
 *
 * [458] Poor Pigs
 */

using System;

public class Solution {
    public int PoorPigs(int buckets, int minutesToDie, int minutesToTest) {
        return (int) Math.Ceiling(Math.Log(buckets, minutesToTest / minutesToDie + 1));
    }
}

